### PR TITLE
[ticket/15258] Adds a method to get a service by class in service_collection

### PR DESCRIPTION
**Pull request by Nicofuma**
PHPBB3-15258

Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15258

### CODE REVIEW

File: phpBB/phpbb/di/service_collection.php [hunk 1]
```diff
@@ -103,4 +103,35 @@ public function get_service_classes()
...
+	{
+		$service_id = null;
+
+		foreach ($this->service_classes as $id => $service_class)
+		{
+			if ($service_class === $class)
+			{
+				if ($service_id !== null)
+				{
+					throw new \RuntimeException('Many services found for class "'.$class.'" in collection.');
```
Comment (CHItA) [inline phpBB/phpbb/di/service_collection.php]: I think `More than one service definitions found for...` would be a bit more descriptive of what the problem really is. Other than this, LGTM.

Comment (Nicofuma) [inline phpBB/phpbb/di/service_collection.php]: fixed

Comment (JimMH) [inline phpBB/phpbb/di/service_collection.php]: It's a bit nitpicking, but that would be singular then: 

> More than one service definition found for ...

Comment (Nicofuma) [inline phpBB/phpbb/di/service_collection.php]: done

**Comment by Nicofuma** [comment]
@marc1706 @CHItA up :)

**Comment by CHItA** [comment]
Btw, do we need this in 3.2?

**Comment by Nicofuma** [comment]
I don't think, do not hesitate to change the target and merge in 3.3 if you prefer

**Pull request merged**